### ISSUE TITLE
Accessibility for DropdownV2

### ISSUE THREAD

**Issue by gracelo** [IBM/carbon-components-react#1280]
Accessibility support for DropdownV2.

My react component uses the DropdownV2 component.  Something like:
```
<DropdownV2 type='inline' label='Label' className='one-of'
             items={oneOfNames.map((label, id) => ({ label, id }))}
             initialSelectedItem={oneOfNames[this.state.oneOfIndex]}
             onChange={this.onOneOfSelection} aria-label='one-of-list'/> 
```

The html generated on browser is:
`<div role="listbox" tabindex="0" class="bx--dropdown one-of bx--list-box bx--list-box--inline"><div role="button" class="bx--list-box__field" tabindex="0" type="button" aria-label="open menu" aria-expanded="false" aria-haspopup="true" data-toggle="true">`...

When run with Chrome DAP accessibility tool, I got error complaining:
"An interactive element/widget must have an accessible name." for the listbox.

Please fix it or advice what to do.

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
"carbon-components": "~9.1.13",
"carbon-components-react": "^6.12.1",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
Develop console

**Comment by joshblack** [comment]
Awesome! Glad it worked 😄

**Comment by gracelo** [comment]
No I meant that the carbon component needs to add this aria-label automatically when user uses DropdownV2.  @joshblack

**Comment by joshblack** [comment]
Hi @gracelo! We can't really supply the `aria-label` because we don't know what you are using the control, if that makes sense. We could only provide a non-descriptive one like "Dropdown" which wouldn't be the best experience for users. What do you think?

**Comment by gracelo** [comment]
sure, let's try that.  Thanks.

**Comment by joshblack** [comment]
Sounds good! Going to close this for now. Feel free to comment if this still persists 👍

**Comment by gracelo** [comment]
@joshblack , I don't understand.    Our code just use the DropdownV2 component.   There is no aria-label generated for` <div role="listbox"... ` and DAP failed.  What can I do?

**Comment by joshblack** [comment]
@gracelo the way I understood the DAP violation is that the element with the listbox role needed an appropriate `aria-label` attribute. Unfortunately, we can't provide that label by default as we don't know what consumers are using the dropdown for. As a result, to resolve the issue you will need to pass in an `aria-label` attribute to `DropdownV2` that will then be put on the node with the `listbox` role and subsequently solve the DAP violation 👍

**Comment by gracelo** [comment]
@joshblack , I think I already pass  aria-label='one-of-list' into the DropdownV2 component but it is not included in the generated html of the 'bx--dropdown' div.  Please check the behaviour.  Thanks.

**Comment by joshblack** [comment]
Gotcha, sounds good! Let me know re-open this and have this issue be for adding support for that in 😄
